src/AWSS3.jl

```diff
@@ -800,6 +778,122 @@ function s3_sign_url(aws::AWSConfig, bucket, path, seconds=3600;
     return "$endpoint/$path?$(HTTP.escapeuri(query))"
 end
 
+
+function _s3_sign_url_v4(
+    aws::AWSConfig, bucket, path, seconds=3600;
+    verb="GET", content_type="application/octet-stream", protocol="http"
+)
+
+    path = HTTP.escapepath("/$bucket/$path")
+
+    now_datetime = now(Dates.UTC)
+    datetime_stamp = Dates.format(now_datetime, "YYYYmmddTHHMMSSZ")
+    date_stamp = Dates.format(now_datetime, "YYYYmmdd")
+
+    service = "s3"
+    scheme = "AWS4"
+    algorithm = "HMAC-SHA256"
+    terminator = "aws4_request"
+
+    scope = "$date_stamp/$(aws[:region])/$service/$terminator"
+    host = string("s3-", aws[:region], ".amazonaws.com")
```

Comment:
```suggestion
    host = string("s3-", aws[:region], ".amazonaws.com")
```

I'm not sure if `s3-` is correct here, should this not be `s3.`?

Reply:
This is from the Java sample code I relied upon
https://gist.github.com/kevin-yen/32183655ad7e0b57bec7d678a51e8260#file-presignedurlsample-java-L32
so I think it's `s3-[region]` for regions outside of us-east-1. I had to double-check that but I believe it is `s3-` and I have it tested the URL in our qa and prod environments so I'm fairly positive it's correct

Reply:
I pulled down this branch and ran the tests locally and is was problematic on this [line](https://github.com/JuliaCloud/AWSS3.jl/blob/master/test/awss3.jl#L73). I think adding in tests to the package will clear this up.

Reply:
I'm getting a different error but I just realized I haven't set up the AWS credentials on the machine I was running the test. I think that might be the issue. Let me get the right credentials on to the machine

Reply:
Yep that was it. I needed to setup AWS credentials on my local computer